tests/test_filter_params.py

```diff
@@ -0,0 +1,562 @@
+import pytest
```

Comment:
I didn't want to make too many testing changes in this PR but ...

I think I might be able to refactor test_api.py and have it use the same set of invalid fields in the entities fixture here. Would probably pull the entities fixture out of here and into somewhere else, maybe conftest? Also may be able to refactor test_pagination.py to use this.